File: ppJoke/libcommon/src/main/java/com/example/libcommon/utils/PixUtils.java
package com.example.libcommon.utils;

import android.util.DisplayMetrics;

/**
 * @author tengfei
 * date 2020-02-08 12:48
 * email arrayadapter.cn@gmail.com
 * description
 */
public class PixUtils {

    public static int dp2px(int dpValue) {
        DisplayMetrics metrics = AppGlobal.getApplication().getResources().getDisplayMetrics();
        return (int) (metrics.density * dpValue + 0.5f);
    }

    public static int getScreenWidth() {
        DisplayMetrics metrics = AppGlobal.getApplication().getResources().getDisplayMetrics();
        return metrics.widthPixels;
    }

    public static int getScreenHeight() {
        DisplayMetrics metrics = AppGlobal.getApplication().getResources().getDisplayMetrics();
        return metrics.heightPixels;
    }
}

File: ppJoke/libcommon/src/main/java/com/example/libcommon/utils/PixUtils.kt
package com.example.libcommon.utils

import android.util.DisplayMetrics


/**
 * @author tengfei
 * date 2020-02-07 23:58
 * email arrayadapter.cn@gmail.com
 * description
 */
fun dp2px(dpValue: Int): Int {
    val metrics: DisplayMetrics = AppGlobal.getApp()!!.resources.displayMetrics
    return (metrics.density * dpValue + 0.5f).toInt()
}

fun getScreenWidth(): Int {
    val metrics: DisplayMetrics = AppGlobal.getApp()!!.resources.displayMetrics
    return metrics.widthPixels
}

fun getScreenHeight(): Int {
    val metrics: DisplayMetrics = AppGlobal.getApp()!!.resources.displayMetrics
    return metrics.heightPixels
}